packages/aws-cdk/lib/api/cxapp/cloud-executable.ts

```diff
@@ -89,54 +90,76 @@ export class CloudExecutable {
         }
       }
 
-      if (trackVersions && assembly.runtime) {
-        const modules = formatModules(assembly.runtime);
-        for (const stack of assembly.stacks) {
-          if (!stack.template.Resources) {
-            stack.template.Resources = {};
-          }
-          const resourcePresent = stack.environment.region === cxapi.UNKNOWN_REGION
-            || RegionInfo.get(stack.environment.region).cdkMetadataResourceAvailable;
-          if (resourcePresent) {
-            if (!stack.template.Resources.CDKMetadata) {
-              stack.template.Resources.CDKMetadata = {
-                Type: 'AWS::CDK::Metadata',
-                Properties: {
-                  Modules: modules,
-                },
-              };
-              if (stack.environment.region === cxapi.UNKNOWN_REGION) {
-                stack.template.Conditions = stack.template.Conditions || {};
-                const condName = 'CDKMetadataAvailable';
-                if (!stack.template.Conditions[condName]) {
-                  stack.template.Conditions[condName] = _makeCdkMetadataAvailableCondition();
-                  stack.template.Resources.CDKMetadata.Condition = condName;
-                } else {
-                  warning(`The stack ${stack.id} already includes a ${condName} condition`);
-                }
-              }
-            } else {
-              warning(`The stack ${stack.id} already includes a CDKMetadata resource`);
-            }
-          }
-        }
+      if (trackVersions) {
+        // @deprecated(v2): remove this 'if' block and all code referenced by it.
+        // This should honestly not be done here. The framework
+        // should (and will, shortly) synthesize this information directly into
+        // the template. However, in order to support old framework versions
+        // that don't synthesize this info yet, we can only remove this code
+        // once we break backwards compatibility.
```

Comment:
I still think we shouldn't add this notice **until** the change is done to the framework